panic: runtime error with use of curl in reporting

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/nuclei/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #nuclei channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version:
v2.5.3

### Current Behavior:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xcbb114]

goroutine 339 [running]:
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).executeRequest(0xc000340dc0, {0xc0000a4120, 0x1c}, 0xc001a87a70, 0x0, 0x0, 0xc0000fbd98, 0xc000b08930)
	/home/runner/work/nuclei/nuclei/v2/pkg/protocols/http/request.go:389 +0x10f4
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).ExecuteWithResults(0xc000340dc0, {0xc0000a4120, 0x1c}, 0xc001a87890, 0xc000b0caf0, 0xc001a878f0)
	/home/runner/work/nuclei/nuclei/v2/pkg/protocols/http/request.go:240 +0x3f3
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0xc000cd5ec0, {0xc0000a4120, 0x1c})
	/home/runner/work/nuclei/nuclei/v2/pkg/protocols/common/executer/executer.go:52 +0x18e
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).processTemplateWithList.func1.1({0xc0000a4120, 0xc000b12b60})
	/home/runner/work/nuclei/nuclei/v2/internal/runner/processor.go:34 +0xa7
created by github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).processTemplateWithList.func1
	/home/runner/work/nuclei/nuclei/v2/internal/runner/processor.go:31 +0x17d
```

### Expected Behavior:
Running without crash
### Steps To Reproduce:

Affected file: `v2/pkg/protocols/http/request.go`

Affected line of code:-

```diff
	var curlCommand string
	if !request.Unsafe && resp != nil && generatedRequest.request != nil {
		bodyBytes, _ := generatedRequest.request.BodyBytes()
+ 		resp.Request.Body = ioutil.NopCloser(bytes.NewReader(bodyBytes))
		command, _ := http2curl.GetCurlCommand(resp.Request)
		if err == nil && command != nil {
			curlCommand = command.String()
		}
	}
```

Reported by Rojan

--- a/v2/pkg/protocols/http/request.go
+++ b/v2/pkg/protocols/http/request.go
@@ -384,7 +384,7 @@ func (request *Request) executeRequest(reqURL string, generatedRequest *generate
 	}()
 
 	var curlCommand string
-	if !request.Unsafe && resp != nil && generatedRequest.request != nil {
+	if !request.Unsafe && resp != nil && generatedRequest.request != nil && resp.Request != nil {
 		bodyBytes, _ := generatedRequest.request.BodyBytes()
 		resp.Request.Body = ioutil.NopCloser(bytes.NewReader(bodyBytes))
 		command, _ := http2curl.GetCurlCommand(resp.Request)

